### PR TITLE
Fixed issue in greeting.yml workflow #874

### LINKED ISSUE #874: [Bug]: Issue in greeting.yml workflow
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

greeting.yml workflow is only for when a contributor create first issue

### Expected behavior

I want to update it so that it can comment on any issue .

### Add ScreenShots

Screenshot 2024-06-09 at 7 49 50 PM

Please let me work on this 

### On which device are you experiencing this bug?

_No response_

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by Anshika14528**
## Related Issue

fix #874

## Description

Updated the greeting.yml worflow.

## Type of PR

- [x] Bug fix
- [ ] Feature enhancement
- [ ] Documentation update
- [ ] Other (specify): _______________

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [ ] I have commented my code, particularly in hard-to-understand areas.

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

File: .github/workflows/greetings.yml [hunk 1]
```diff
@@ -1,20 +1,23 @@
-name: Greetings
+name: 'Greetings'
```
Comment (Ajay-Dhangar) [inline .github/workflows/greetings.yml:1]: Replace your code to

```yml
name: 'Greetings'

on:
  issues:
    types: [opened]
  pull_request_target:
    types: [opened]

permissions:
  issues: write
  pull-requests: write

jobs:
  welcome:
    runs-on: ubuntu-latest

    steps:
      - name: Check out repository
        uses: actions/checkout@v2

      - name: Greet first-time contributors
        id: greet
        uses: actions/first-interaction@v1
        with:
          repo-token: ${{ secrets.GITHUB_TOKEN }}
          issue-message: |
            Hi there! 👋 Thank you for opening your first issue on CodeHarborHub. We're excited to hear your thoughts and help you out. Please provide as much detail as you can so we can assist you effectively.
          pr-message: |
            Hi there! 👋 Thank you for submitting your first pull request to CodeHarborHub. We appreciate your contribution! Our team will review it soon. If you have any questions or need further assistance, feel free to reach out.

      - name: Assign issue or pull request to team member
        if: github.event_name == 'issues' || github.event_name == 'pull_request_target'
        run: |
          ISSUE_NUMBER=${{ github.event.issue.number || github.event.pull_request.number }}
          curl -X POST -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
          -d '{"assignees":["team-member"]}' \
          "https://api.github.com/repos/${{ github.repository }}/issues/${ISSUE_NUMBER}"

      - name: Welcome message for community contributors
        if: github.event_name == 'issues' || github.event_name == 'pull_request_target'
        uses: EddieHubCommunity/gh-action-community/src/welcome@main
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}
          issue-message: "Hi there! Thanks for opening this issue. We appreciate your contribution to this open-source project. We aim to respond or assign your issue as soon as possible."
          pr-message: "Great job, @${{ github.actor }}! 🎉 Thank you for submitting your pull request to CodeHarborHub. We appreciate your contribution! Our team will review it soon. If you have any questions or need further assistance, feel free to reach out."

      - name: Label first-time contributions
        if: github.event_name == 'issues' || github.event_name == 'pull_request_target'
        run: |
          ISSUE_NUMBER=${{ github.event.issue.number || github.event.pull_request.number }}
          curl -X POST -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
          -d '{"labels":["first-time contribution"]}' \
          "https://api.github.com/repos/${{ github.repository }}/issues/${ISSUE_NUMBER}"
```

for performing all task

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by Anshika14528** [comment]
Done please review it now @Ajay-Dhangar sir

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**